docs/spec.md

```diff
@@ -4663,12 +4663,13 @@ Given that, `results = exec(schedule, inputs)`, where:
 * `schedule` is an implementation-defined permutation of the index space
   of `updates[0]`.
 * `exec([update_index, ...], results) = exec([...], updated_results)` where:
-  * `updated_values =
-    update_computation(results[:][result_index], updates[:][update_index])`.
-  * `updated_results` is a copy of `results` with `results[:][result_index]`
-    set to `updated_values[:]`.
-  * If `result_index` is out of bounds for `shape(results[:])`, the behavior
-    is implementation-defined.
+  * If `result_index` is in bounds for `shape(results[:])`
+    * `updated_values =
+      update_computation(results[:][result_index], updates[:][update_index])`.
+    * `updated_results` is a copy of `results` with `results[:][result_index]`
+      set to `updated_values[:]`.
+  * Otherwise
+    * `updated_results = results`.
 * `exec([], results) = results`.
 
 If `indices_are_sorted` is `true` then the implementation can assume that
```

Comment:
Let's also update the image similarly to #1541? We could similarly change `0 | 2` to `0 | 9`, but unlike with GatherOp, we'd need to recolor one of the green rectangles into white, and I would also drop the green highlighting from the corresponding index: https://docs.google.com/presentation/d/1vr6x3VtV77NPIwR26uiHEJQb407QtRXMUfIi5CSs8aA/edit#slide=id.g24e7ed57a8e_0_5.

Reply:
I'd also propose that we update `interpret_scatter.mlir` accordingly and drop the second test, similarly to what's done for GatherOp in #1541.